src/components/VendorDashboard/VendorDashboard.tsx

```diff
@@ -5,59 +5,130 @@ import { getWeddings } from '../../apiCalls'
 import { Link } from 'react-router-dom';
 import { StyledButton } from '../App/styledComponents.styles'
 import './VendorDashboard.css';
+import dayjs from 'dayjs';
 
 type Wedding = {
   id: number;
   name: string;
   email: string;
-  date: string;
+  date: any;
   image: string;
 }
 
 function VendorDashboard() {
   const [weddings, setWeddings] = useState<Wedding[]>([])
+  const [weddingFilter, setWeddingFilter] = useState<Wedding[]>([])
+  const [errorMessage, setErrorMessage] = useState('')
+  const [hasError, setError] = useState(false)
 
   useEffect(() => {
-  const allWeddings = async () => {
-    const result = await getWeddings()
-    console.log(result)
-    setWeddings(result)
+    const allWeddings = async () => {
+      const result = await getWeddings()
+      if(result.length) {
+        result.forEach((wed: any) => {
+          wed.date = new Date(wed.date)
+        })
+        let sortedResult = result.sort((a: any, b: any) => a.date - b.date)
+        setWeddings(sortedResult)
+      }
+    }
+    allWeddings()
+  }, [])
+
+  const filterWeddings = (e: any) => {
+    setError(false)
+    setErrorMessage('')
+    let sortedFilter
+    switch(e.target.options.selectedIndex) {
+      case 1:
+        const futureWeddings = weddings.filter((a: any) => a.date > Date.now())
+        if(futureWeddings.length) {
+          sortedFilter = futureWeddings.sort((a: any, b: any) => a.date - b.date)
+          setWeddingFilter(sortedFilter)
+        } else {
+          setError(true)
+          setErrorMessage('No Upcoming Weddings To Show')
+        }
+        break;
+      case 2:
+        const pastWeddings = weddings.filter((a: any) => a.date < Date.now())
+        if(pastWeddings.length) {
+          sortedFilter = pastWeddings.sort((a: any, b: any) => a.date - b.date)
+          setWeddingFilter(sortedFilter)
+        } else {
+          setError(true)
+          setErrorMessage('No Past Weddings To Show')
+        }
+        break;
+      case 3:
+        const currentWeddings = weddings.filter((a: any) => a.date == Date.now())
+        if(currentWeddings.length) {
+          sortedFilter = currentWeddings.sort((a: any, b: any) => a.date - b.date)
+          setWeddingFilter(sortedFilter)
+        } else {
+          setError(true)
+          setErrorMessage('No Weddings Taking Place Today')
+        }
+        break;
+      default:
+        setWeddings(weddings)
+        setWeddingFilter([])
+        break;
```

Comment:
Interesting conditional here. Glad you got it figured out. This is a complicated thing to think/plan through

Reply:
Seems like a good place to use switch statements!